test/views/declaration/CarrierDetailsViewSpec.scala

```diff
@@ -314,16 +296,15 @@ class CarrierDetailsViewSpec extends UnitViewSpec with CommonMessages with Stubs
             .fillAndValidate(CarrierDetails(EntityDetails(None, Some(Address("Marco Polo", "", "", "", "")))))
         )
 
-        view must haveGlobalErrorSummary
-        view must haveFieldErrorLink("details_address_addressLine", "#details_address_addressLine")
-        view must haveFieldErrorLink("details_address_townOrCity", "#details_address_townOrCity")
-        view must haveFieldErrorLink("details_address_postCode", "#details_address_postCode")
-        view must haveFieldErrorLink("details_address_country", "#details_address_country")
-
-        view.getElementById("error-message-details_address_addressLine-input").text() mustBe messages("supplementary.address.addressLine.empty")
-        view.getElementById("error-message-details_address_townOrCity-input").text() mustBe messages("supplementary.address.townOrCity.empty")
-        view.getElementById("error-message-details_address_postCode-input").text() mustBe messages("supplementary.address.postCode.empty")
-        view.select("span.error-message").text() mustBe messages("supplementary.address.country.empty")
+        view must haveGovukGlobalErrorSummary
+        view must containErrorElementWithTagAndHref("a", "#details_address_addressLine")
+        view.getElementById("details_address_addressLine-error") must containMessage("supplementary.address.addressLine.empty")
```

Comment:
Did you consider the equivalent approach `view must containErrorElementWithMessage(s"supplementary.address.addressLine.empty")` ?